pkg/resource/hosted_zone/hook.go

```diff
@@ -13,3 +18,177 @@ import (
 func getCallerReference(zone *svcapitypes.HostedZone) string {
 	return fmt.Sprintf("%s-%d", *zone.Spec.Name, time.Now().UnixMilli())
 }
+
+func (rm *resourceManager) customUpdateHostedZone(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+	delta *ackcompare.Delta,
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.customUpdateHostedZone")
+	defer exit(err)
+
+	// Default `updated` to `desired` because it is likely
+	// EC2 `modify` APIs do NOT return output, only errors.
+	// If the `modify` calls (i.e. `sync`) do NOT return
+	// an error, then the update was successful and desired.Spec
+	// (now updated.Spec) reflects the latest resource state.
+	updated = rm.concreteResource(desired.DeepCopy())
+
+	if delta.DifferentAt("Spec.Tags") {
+		if err := rm.syncTags(ctx, desired, latest); err != nil {
+			return nil, err
+		}
+	}
+
+	return updated, nil
+}
+
+// syncTags used to keep tags in sync by calling Create and Delete API's
+func (rm *resourceManager) syncTags(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.syncTags")
+	defer func(err error) {
+		exit(err)
+	}(err)
+
+	resourceId := latest.ko.Status.ID
+
+	toAdd, toDelete := computeTagsDelta(
+		desired.ko.Spec.Tags, latest.ko.Spec.Tags,
+	)
+
+	resourceType := "hostedzone"
+	if len(toDelete) > 0 {
+		rlog.Debug("removing tags from HostedZone resource", "tags", toDelete)
+		_, err = rm.sdkapi.ChangeTagsForResource(
+			&svcsdk.ChangeTagsForResourceInput{
+				ResourceId:    resourceId,
+				RemoveTagKeys: toDelete,
+				ResourceType:  &resourceType,
+			},
+		)
+		rm.metrics.RecordAPICall("UPDATE", "DeleteTags", err)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	if len(toAdd) > 0 {
+		rlog.Debug("adding tags to HostedZone resource", "tags", toAdd)
+		_, err = rm.sdkapi.ChangeTagsForResource(
+			&svcsdk.ChangeTagsForResourceInput{
+				ResourceId:   resourceId,
+				AddTags:      rm.sdkTags(toAdd),
+				ResourceType: &resourceType,
+			},
+		)
+		rm.metrics.RecordAPICall("UPDATE", "CreateTags", err)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// sdkTags converts *svcapitypes.Tag array to a *svcsdk.Tag array
+func (rm *resourceManager) sdkTags(
+	tags []*svcapitypes.Tag,
+) (sdktags []*svcsdk.Tag) {
+
+	for _, i := range tags {
+		sdktag := rm.newTag(*i)
+		sdktags = append(sdktags, sdktag)
+	}
+
+	return sdktags
+}
+
+// computeTagsDelta returns tags to be added and removed from the resource
+func computeTagsDelta(
```

Comment:
I'm trying to push people to use [`GetTagsDifference`](https://github.com/aws-controllers-k8s/ec2-controller/blob/55cbd30c3eb4ee23c7ba35766320451caf46648a/pkg/resource/network_acl/hooks.go#L46-L52) wherever possible, so we don't have different comparison implementations everywhere in the code. Could you please use this for your use case?

Reply:
done